weather_mv/loader_pipeline/sinks.py

```diff
@@ -394,29 +424,42 @@ def open_dataset(uri: str,
             return
         with open_local(uri) as local_path:
             _, uri_extension = os.path.splitext(uri)
-            xr_dataset: xr.Dataset = __open_dataset_file(local_path,
-                                                         uri_extension,
-                                                         disable_grib_schema_normalization,
-                                                         open_dataset_kwargs)
-            if uri_extension in ['.tif', '.tiff']:
-                xr_dataset = _preprocess_tif(xr_dataset,
+            xr_datasets: xr.Dataset = __open_dataset_file(local_path,
+                                                          uri_extension,
+                                                          disable_grib_schema_normalization,
+                                                          open_dataset_kwargs,
+                                                          group_common_hypercubes)
+            # Extracting dtype, crs and transform from the dataset.
+            with rasterio.open(local_path, 'r') as f:
+                dtype, crs, transform = (f.profile.get(key) for key in ['dtype', 'crs', 'transform'])
+
+            if group_common_hypercubes:
+                total_size_in_bytes = 0
+
+                for xr_dataset in xr_datasets:
+                    xr_dataset.attrs.update({'dtype': dtype, 'crs': crs, 'transform': transform})
+                    total_size_in_bytes += xr_dataset.nbytes
+
+                logger.info(f'opened dataset size: {total_size_in_bytes}')
+            elif uri_extension in ['.tif', '.tiff']:
+                xr_dataset = _preprocess_tif(xr_datasets,
                                              local_path,
                                              tif_metadata_for_datetime,
                                              uri,
                                              band_names_dict,
                                              initialization_time_regex,
                                              forecast_time_regex)
+            else:
+                xr_dataset = xr_datasets
 
-            # Extracting dtype, crs and transform from the dataset & storing them as attributes.
-            with rasterio.open(local_path, 'r') as f:
-                dtype, crs, transform = (f.profile.get(key) for key in ['dtype', 'crs', 'transform'])
-                xr_dataset.attrs.update({'dtype': dtype, 'crs': crs, 'transform': transform})
+            xr_dataset.attrs.update({'dtype': dtype, 'crs': crs, 'transform': transform})
```

Comment:
Looks off to me. Let's discuss this over call.

Reply:
Resolved